### PR TITLE
Workaround for ESLint plugin module resolution

### PR DESCRIPTION
**Pull request by tylerbutler**
I was sorely mistaken in #4802 that ESLint 7 fully addressed their plugin module resolution logic. It doesn't; see https://github.com/eslint/eslint/issues/3458.

This change adds a workaround using [@rushstack/eslint-patch](https://www.npmjs.com/package/@rushstack/eslint-patch). A future change will upgrade all packages to use this config.

### CODE REVIEW

File: common/build/eslint-config-fluid/eslint7.js [hunk 1]
```diff
@@ -32,6 +32,9 @@
   }
 */
 
+// // This is a workaround for https://github.com/eslint/eslint/issues/3458
```
Comment (DLehenbauer) [inline common/build/eslint-config-fluid/eslint7.js:35]: Nit: Commented comment.  :-)

File: common/build/eslint-config-fluid/eslint7.js [hunk 2]
```diff
@@ -214,7 +217,6 @@ module.exports = {
         ],
 
         // eslint-plugin-unicorn
-        "unicorn/better-regex": "error",
```
Comment (DLehenbauer) [inline common/build/eslint-config-fluid/eslint7.js:217]: Double check that you intended to remove this config?

File: common/build/eslint-config-fluid/patch/modern-module-resolution.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+// This is provided so packages that depend on this config don't have to add @rushstack/eslint-patch as
+// a dependency
+require("@rushstack/eslint-patch/modern-module-resolution");
```
Comment (DLehenbauer) [inline common/build/eslint-config-fluid/patch/modern-module-resolution.js:8]: Interesting convention.

**Review by DLehenbauer — APPROVED** [review]

**Comment by DLehenbauer** [comment]
I've use 'eslint-patch' in Rush monorepos and it worked as advertised.

You might need to disable Lerna hoisting for the ESLint plugins, depending on if resolving "relative to the local folder" includes parent folders.  (I don't know...  just something that's tripped tools up in the past.)

File: common/build/eslint-config-fluid/eslint7.js (hunk 1 above)
Comment (tylerbutler) [inline common/build/eslint-config-fluid/eslint7.js:35]:
```suggestion
// This is a workaround for https://github.com/eslint/eslint/issues/3458
```

File: common/build/eslint-config-fluid/eslint7.js (hunk 2 above)
Comment (tylerbutler) [reply common/build/eslint-config-fluid/eslint7.js:217]: Oops!

**Comment by tylerbutler** [comment]
> I've use 'eslint-patch' in Rush monorepos and it worked as advertised.
> 
> You might need to disable Lerna hoisting for the ESLint plugins, depending on if resolving "relative to the local folder" includes parent folders. (I don't know... just something that's tripped tools up in the past.)

Thanks! It's been challenging to test all this effectively because of the hoisting and module resolution logic that works when it shouldn't. :)

**Pull request merged**